src/cpu/flexcpu/flexcpu_thread.cc

```diff
@@ -120,18 +128,18 @@ FlexCPUThread::activate()
 void
 FlexCPUThread::advanceInst(TheISA::PCState next_pc)
 {
-    if (status() != ThreadContext::Active) {
-        // If this thread isn't currently active, don't do anything
-        return;
-    }
-
     advanceInstNeedsRetry = static_cast<bool>(inflightInstsMaxSize)
                          && inflightInsts.size() >= inflightInstsMaxSize;
     if (advanceInstNeedsRetry) {
         advanceInstRetryPC = next_pc;
         return;
     }
 
+    if (status() != ThreadContext::Active) {
```

Comment:
Why is this now after the retry? This doesn't make sense to me. If it's not active, why would we fetch?

Reply:
Can't think of why either. It might be an artifact of the rebasing that was done when reordering commits. I'm going to remove these patches from the commit.